pkg/debug/checkups/tuf.go

```diff
@@ -54,51 +58,160 @@ func (tc *tufCheckup) Run(ctx context.Context, extraFH io.Writer) error {
 		return err
 	}
 
-	response, err := tc.fetchTufVersion(httpClient, tufUrl)
+	// Primarily, we want to validate that we can talk to the TUF server
+	releaseVersion, remoteMetadataVersion, err := tc.remoteTufMetadata(httpClient, tufUrl)
 	if err != nil {
 		tc.status = Erroring
 		tc.data[tufUrl.String()] = err.Error()
 		tc.summary = "Unable to gather tuf version response"
 		return nil
 	}
 
-	if response == "" {
+	if releaseVersion == "" {
 		tc.status = Failing
 		tc.data[tufUrl.String()] = "missing from tuf response"
 		tc.summary = "missing version from tuf targets response"
 		return nil
 	}
 
 	tc.status = Passing
-	tc.data[tufUrl.String()] = response
-	tc.summary = fmt.Sprintf("Successfully gathered release version %s from %s", response, tufUrl.String())
+	tc.data[tufUrl.String()] = releaseVersion
+	tc.summary = fmt.Sprintf("Successfully gathered release version %s from %s", releaseVersion, tufUrl.String())
+
+	// Gather additional data only if we're running flare
+	if extraFH == io.Discard {
+		return nil
+	}
+
+	tufData := map[string]any{
+		"remote_metadata_version": remoteMetadataVersion,
+	}
+
+	if localMetadataVersion, err := tc.localTufMetadata(); err != nil {
+		tufData["local_metadata_version"] = fmt.Sprintf("error getting local metadata version: %v", err)
+	} else {
+		tufData["local_metadata_version"] = localMetadataVersion
+	}
+
+	if localLauncherVersions, err := tc.versionsInLauncherLibrary(); err != nil {
+		tufData["launcher_versions_in_library"] = fmt.Sprintf("error getting versions available in local launcher library: %v", err)
+	} else {
+		tufData["launcher_versions_in_library"] = localLauncherVersions
+	}
+
+	if localOsqueryVersions, err := tc.versionsInOsquerydLibrary(); err != nil {
+		tufData["osqueryd_versions_in_library"] = fmt.Sprintf("error getting versions available in local osqueryd library: %v", err)
+	} else {
+		tufData["osqueryd_versions_in_library"] = localOsqueryVersions
+	}
+
+	tufData["selected_versions"] = tc.selectedVersions()
+
+	if b, err := json.Marshal(tufData); err == nil {
+		_, _ = extraFH.Write(b)
+	}
```

Comment:
```suggestion
	if err := json.NewEncoder(extraFH).Encode(tufData); err != nil
		tc.summary += " cannot write extra data"
	}
```

maybe? Or drop the error handling from that.